converter/grpc_interceptor.go

```diff
@@ -0,0 +1,268 @@
+// The MIT License
+//
+// Copyright (c) 2022 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package converter
+
+import (
+	"context"
+
+	"go.temporal.io/api/command/v1"
+	commonpb "go.temporal.io/api/common/v1"
+	enumspb "go.temporal.io/api/enums/v1"
+	"go.temporal.io/api/failure/v1"
+	historypb "go.temporal.io/api/history/v1"
+	"go.temporal.io/api/workflowservice/v1"
+	"google.golang.org/grpc"
+)
+
+type serviceInterceptor struct {
+	encoders []PayloadEncoder
+}
+
+func (s *serviceInterceptor) encodePayloads(payloads ...*commonpb.Payloads) error {
+	for _, p := range payloads {
+		for _, payload := range p.GetPayloads() {
+			for i := len(s.encoders) - 1; i >= 0; i-- {
+				if err := s.encoders[i].Encode(payload); err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func (s *serviceInterceptor) decodePayloads(payloads ...*commonpb.Payloads) error {
+	for _, p := range payloads {
+		for _, payload := range p.GetPayloads() {
+			for _, encoder := range s.encoders {
+				if err := encoder.Decode(payload); err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func (s *serviceInterceptor) processRequest(req interface{}) error {
+	switch r := req.(type) {
```

Comment:
Missing `QueryWorkflowRequest.query.query_args` and `RespondWorkflowTaskCompletedRequest.query_results.*.answer`